kortex_driver/src/hardware_interface.cpp

```diff
@@ -841,76 +818,66 @@ void KortexMultiInterfaceHardware::readGripperPosition()
 return_type KortexMultiInterfaceHardware::write(
   const rclcpp::Time & /*time*/, const rclcpp::Duration & /*period*/)
 {
-  if (block_write)
+  if (in_fault_)
   {
-    feedback_ = base_cyclic_.RefreshFeedback();
-    return return_type::OK;
+    RCLCPP_ERROR(LOGGER, "Error! The hardware is in a fault state, unable to write commands.");
+    return hardware_interface::return_type::ERROR;
   }
 
-  if (!std::isnan(reset_fault_cmd_) && fault_controller_running_)
+  if (block_write)
   {
-    resetFaults();
+    feedback_ = base_cyclic_.RefreshFeedback();
+    return return_type::OK;
   }
 
-  if (in_fault_ == 0.0)
+  // Twist controller active?
+  if ((arm_mode_ == k_api::Base::ServoingMode::SINGLE_LEVEL_SERVOING))
   {
-    if (arm_mode_ == k_api::Base::ServoingMode::SINGLE_LEVEL_SERVOING)
+    if (twist_controller_running_)
     {
-      // Twist controller active
-      if (twist_controller_running_)
-      {
-        // twist control
-        sendTwistCommand();
-      }
-      else
-      {
-        // Keep alive mode - no controller active
-        RCLCPP_DEBUG(LOGGER, "No controller active in SINGLE_LEVEL_SERVOING mode!");
-      }
+      // twist control
+      sendTwistCommand();
 
       // gripper control
       sendGripperCommand(
         arm_mode_, gripper_command_position_, gripper_speed_command_, gripper_force_command_);
       // read after write in twist mode
       feedback_ = base_cyclic_.RefreshFeedback();
     }
-    else if (
-      (arm_mode_ == k_api::Base::ServoingMode::LOW_LEVEL_SERVOING) &&
-      (feedback_.base().active_state() == k_api::Common::ARMSTATE_SERVOING_LOW_LEVEL))
+    else
     {
-      // Per joint controller active
-
+      // Keep alive mode - no controller active
+      RCLCPP_DEBUG(LOGGER, "No controller active in SINGLE_LEVEL_SERVOING mode!");
```

Comment:
Can we please throttle this message. It fills up the logs when the arm comes up.